service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/twin/TwinClient.java

```diff
@@ -321,11 +321,11 @@ public Twin replace(Twin twin, String ifMatch) throws IotHubException, IOExcepti
         URL url;
         if (twin.getModuleId() == null || twin.getModuleId().length() == 0)
         {
-            url = this.iotHubConnectionString.getUrlTwin(twin.getDeviceId());
+            url = IotHubConnectionString.getUrlTwin(this.hostName, twin.getDeviceId());
```

Comment:
Does this method exist both as a static method and instance method on IotHubConnectionString? If yes, we should probably rename them to highlight their difference?